src/aequitas/bias.py

```diff
@@ -666,59 +711,95 @@ def _get_statistical_significance(cls, original_df, disparity_df, ref_dict,
+                # original_df.loc[:, binary_col] = original_df.loc[:, SIGNIF_BASES[binary_col]]
+                original_df.loc[:, col] = original_df.loc[:, SIGNIF_BASES[col]]
+
+            # original_df.loc[:, 'binary_precision'] = original_df.loc[:, 'binary_fpr']
+            # original_df.loc[:, 'binary_tnr'] = original_df.loc[:, 'binary_fpr']
+            # original_df.loc[:, 'binary_fdr'] = original_df.loc[:, 'binary_fpr']
+            #
+            # # npv, for, tpr are based on false negatives
+            # original_df.loc[:, 'binary_npv'] = original_df.loc[:, 'binary_fnr']
+            # original_df.loc[:, 'binary_tpr'] = original_df.loc[:, 'binary_fnr']
+            # original_df.loc[:, 'binary_for'] = original_df.loc[:, 'binary_fnr']
+            # # fdr numerator matches fpr numerator
+            #
+            # # pprev and ppr based on score
+            # original_df.loc[:, 'binary_ppr'] = original_df.loc[:, 'binary_score']
+            # original_df.loc[:, 'binary_pprev'] = original_df.loc[:, 'binary_score']
+
+
+            # ensure only predicted positive values included in true positive
+            # error based metrics, and only predicted negative values in false
+            # positive error based metrics
+            POSITIVE_ONLY = ['binary_fpr', 'binary_tnr', 'binary_precision', 'binary_fdr']
+            NEGATIVE_ONLY = ['binary_fnr', 'binary_tpr', 'binary_npv', 'binary_for']
+
+            FPR_BASED = [metric for metric in POSITIVE_ONLY if metric in binary_inclusions]
+            FNR_BASED = [metric for metric in NEGATIVE_ONLY if metric in binary_inclusions]
+
+            original_df.loc[original_df['binary_score'] == 0, FPR_BASED] = pd.np.nan
+            original_df.loc[original_df['binary_score'] == 1, FNR_BASED] = pd.np.nan
 
             measures = list(original_df.columns[original_df.columns.str.contains('binary_')])
             measures += ['label_value']
 
             for measure in measures:
-                cls._calculate_significance(
-                    original_df, disparity_df, attribute, measure,
-                    ref_dict=ref_dict,
-                    alpha=alpha)
+                # only calculate significance if in selected_significance
+                if (measure in selected_significance) or (''.join(measure.split('binary_')) in selected_significance):
+
+                    cls._calculate_significance(original_df, disparity_df,
+                                                attribute, measure, ref_dict=ref_dict,
+                                                alpha=alpha)
         return disparity_df
 
+
     def list_disparities(self, df):
         """
         View list of all calculated disparities in df
         """
-        return list(df.columns[df.columns.str.contains('_disparity')])
+        try:
+            return list(df.columns[df.columns.str.contains('_disparity')])
+        except KeyError:
+            raise Exception("No disparity columns found in dataframe. Tip: "
```

Comment:
it's kind of sad to lose our nice and specific exception and replace it with a generic one. to add your custom message, perhaps do so with a `LookupError` or a `KeyError`.